pkg/controller/proxyconfig/validation.go

```diff
@@ -231,9 +233,26 @@ func validateReadinessEndpointWithRetries(caBundle []*x509.Certificate, proxy, e
 // returns an error if a 2XX or 3XX http status code is not returned.
 // caBundle is used to authenticate endpoint if endpoint contains an
 // https scheme.
-func runReadinessProbe(caBundle []*x509.Certificate, proxy, endpoint *url.URL) error {
+func runReadinessProbe(caBundle []*x509.Certificate, proxyURL, endpoint *url.URL) error {
 	transport := &http.Transport{
-		Proxy: http.ProxyURL(proxy),
+		Proxy: http.ProxyURL(proxyURL),
+	}
+
+	if platform.NewHyperShiftConfig().Enabled {
```

Comment:
it is a shame that `func hyperShiftEnabled() bool` is not public. But I think this work around (of getting the config and looking at Enabled) is fine for now.

Reply:
we use the same approach across CNO so I would leave it as is for now.